src/buffer/out/TextColor.h

```diff
@@ -44,7 +44,9 @@ enum class ColorType : BYTE
     IsDefault,
     IsIndex16,
     IsIndex256,
-    IsRgb
+    IsIndex256Colon,
+    IsRgb,
+    IsRgbColon

```

Comment:
@lhecker You mentioned some optimization above the `ColorType` declaration, does the addition of new options negatively affects it in any way? 

I assume if contiguous values are Or'd, it gets compiled as range check by the compiler, right? Then there shouldn't be any effect, since I added it after index256. I suppose the same optimization should also apply to modified `IsRgb()` and `IsIndex256()`, since they basically does the same thing.

Reply:
Eh. In this case, I think we don't need to remember what _shape_ of color the application sent us. 

We talked about this a bit over in #15706 -- since ConPTY cannot detect what kind of terminal emulator it is itself talking to, it should use the "most compatible" shape of color control sequence.

Reply:
(That lets you simplify the changes to TextColor by, well, not having them! :grin:)

Reply:
UGH IGNORE ME I MISREAD THE PR.

Okay so. Okay. _Hmm_. I have no idea how to handle this properly for `DECRQSS`.

Reply:
`DECRQSS` should _always_ return colons for the ODA colors once we support them. There's no need to track what format of sequence was used to set the color.

Reply:
That makes this a MUCH easier discussion 😅

Reply:
> DECRQSS should always return colons for the ODA colors once we support them.

So we should use ODA's format for all `IsIndex256` and `IsRgb` colors in `DECRQSS`?

(I might have not understood what ODA colors are 😅)

Reply:
One thing that still confuses me is,

as @DHowett mentioned:
>  We should accept as much as we can and begrudgingly emit the most compatible thing we can.

But this:
> DECRQSS should always return colons for the ODA colors once we support them.

Is `DECRQSS` expected to be used by newer conpty clients only? (that support ODA colors) But I'm not sure if that's the case, or am I missing something?

Even after we get passthrough mode, WT still needs to talk to older VT clients, and is expected to respond back with `;` as the parameter separator.

Reply:
I guess we can also just use semi-colons for `Fg` and `Bg` colours in the response. It's simple, and we are not losing anything here. Even if feature testing fails, and application assumes we don't support ODA colours, it's expected that application will fallback to using semi-colon instead.

Reply:
When it comes to implementing escape sequences, the conpty client is almost entirely irrelevant. Conhost handles almost all VT sequence processing, stores the results in a buffer, and then forwards a representation of that buffer to the conpty client in a very dumbed-down format.

So when it comes to ODA colors, conhost is the only one that gets to see the original sequence from the app, and thus needs to support both colons and semicolons. What we forward to the conpty client is just the dumbed-down version, which is semicolons. In some cases we may even convert down to 16 colors for primitive clients.

Conhost is also responsible for all VT queries, or at least all the ones we support, and we should really be blocking everything else (any query that the conpty client responds to is going to be meaningless, and this already causes problems for applications). So when it comes to `DECRQSS`, conhost needs to respond with a representation matching the format that it supports, i.e. colons. The conpty client should never see that query.

If we ever get around to making passthrough mode work, then things will be different, but that still doesn't change anything about this implementation. We shouldn't be intercepting the sequences at all, so the conpty client should get exactly what the application sends, and the conpty client would also then be responsible for handling queries like `DECRQSS`. If they don't support colons, then they won't respond to `DECRQSS` with colons, and apps will know not to use them.

Reply:
Ohh, so sorry! 😬 I think I mixed up some terms there. What I actually wanted to know was:

> Is `DECRQSS` only expected to be used by newer ~conpty clients~ VT applications (Eg. shell, vim, neovim) that support ODA colors?

And sorry if I asking too many questions here, but I just want to make sure we are not breaking existing apps that make queries to the conhost. They shouldn't recieve colons when they don't support them, or is it ok to do this?

Reply:
> I just want to make sure we are not breaking existing apps that make queries to the conhost. 

That's a good point, but I don't think it should be a problem in practice, because this is way other terminals already work, at least the few that support querying ODA colors with `DECRQSS`. Have a look at XTerm or Mintty, for example. Regardless of whether you use semicolons or colons when setting the colors, their `DECRQSS` response will always include colons.

And traditionally the main use-case for `DECRQSS` is saving and restoring state. For that an app shouldn't need to parse anything anyway - they just save the returned string and replay it with a `CSI` prefix to restore the attributes to what they were before.

The other more recent use-case is for detecting things like ODA color support. But in that case the app should definitely be expecting to potentially see colons in the response, because that's the whole point of executing the query in the first place.

Reply:
Right on time. I have pushed the requested changes you can have a look.

And about the brokenness of some apps, I realised that support for `DECRQSS` was recently added, so there shouldn't be any issues on that (for us) 👍

Reply:
I believe this can be resolved now. Thanks for the technical details @j4james